plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1288,9 +1301,67 @@ public boolean configure(final String name, final Map<String, Object> params) th
             s_logger.info("iscsi session clean up is disabled");
         }
 
+        setupMemoryBalloonStatsPeriod(conn);
+
         return true;
     }
 
+    /**
+     * Sets the balloon driver of each VM to get the memory stats at the time interval defined in the agent.properties file.
+     * @param conn the Libvirt connection.
+     */
+    protected void setupMemoryBalloonStatsPeriod(Connect conn) {
+        Integer[] vmIds = null;
+        try {
+            vmIds = ArrayUtils.toObject(conn.listDomains());
+        } catch (final LibvirtException e) {
+            s_logger.error("Unable to get the list of Libvirt domains on this host.", e);
+            return;
+        }
+        List<Integer> vmIdList = Arrays.asList(vmIds);
+        s_logger.debug(String.format("We have found a total of [%s] VMs (Libvirt domains) on this host: [%s].", vmIdList.size(), vmIdList.toString()));
+
+        if (vmIdList.isEmpty()) {
+            s_logger.debug("Skipping the memory balloon stats period setting, since there are no VMs (active Libvirt domains) on this host.");
+            return;
+        }
+
+        if (AgentPropertiesFileHandler.getPropertyValue(AgentProperties.VM_MEMBALLOON_DISABLE)) {
+            s_logger.debug(String.format("Skipping the memory balloon stats period setting because the [%s] property is set to 'true'.",
+                    AgentProperties.VM_MEMBALLOON_DISABLE.getName()));
+            return;
+        }
+
+        Integer currentVmBalloonStatsPeriod = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.VM_MEMBALLOON_STATS_PERIOD);
+
+        for (Integer vmId : vmIdList) {
+            String getMemBalloonFromXmlFileCommand = String.format(COMMAND_GET_MEM_BALLOON_FROM_XML_FILE, vmId);
```

Comment:
it is better to update `LibvirtDomainXMLParser` to support `GuestResourceDef` , instead of 
`COMMAND_GET_MEM_BALLOON_FROM_XML_FILE = "virsh dumpxml %s | grep '<memballoon model='";`